lingua_franca/lang/format_fa.py

```diff
@@ -0,0 +1,306 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2017 Mycroft AI Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from lingua_franca.lang.format_common import convert_to_mixed_fraction
+from lingua_franca.lang.common_data_fa import \
+    _FARSI_ONES, _FARSI_TENS, _FARSI_HUNDREDS, _FARSI_BIG, _FARSI_SEPERATOR, \
+    _FARSI_FRAC, _FARSI_FRAC_BIG, _FRACTION_STRING_FA, _FORMAL_VARIANT
+import math
+from lingua_franca.internal import lookup_variant
+from enum import IntEnum
+from functools import wraps
+
+class NumberVariantFA(IntEnum):
+    CONVERSATIONAL = 0
+    FORMAL = 1
+
+lookup_number = lookup_variant({
+    "default": NumberVariantFA.CONVERSATIONAL,
+    "conversational": NumberVariantFA.CONVERSATIONAL,
+    "formal": NumberVariantFA.FORMAL,
+})
+
+def _applyVariant(text, variant):
+    if variant == NumberVariantFA.FORMAL:
+        print("Doing")
+        print(text)
+        for key, value in _FORMAL_VARIANT.items():
+            print("xxx "+value+" "+key)
+            text = text.replace(value, key)
+            print(text)
+    return text
+
+def _handleVariant(func):
```

Comment:
can this be renamed `_handle_number_variant`? makes the method name explicit and uses snake_case for consistency